osbs/utils/__init__.py

```diff
@@ -158,11 +158,28 @@ def clone_git_repo(git_url, target_dir=None, commit=None, retry_times=GIT_MAX_RE
     :param retry_times: int, number of retries for git clone
     :param branch: str, optional branch of the commit, required if depth is provided
     :param depth: int, optional expected depth
+    :param exists_ok: bool, optional whether or not to use a repo if it already exists
     :return: str, int, commit ID of HEAD
     """
     retry_delay = GIT_BACKOFF_FACTOR
     target_dir = target_dir or os.path.join(tempfile.mkdtemp(), "repo")
     commit = commit or "master"
+
+    if os.path.isdir(target_dir) and len(os.listdir(target_dir)) != 0:
+        if exists_ok:
+            try:
+                logger.info("'%s' directory already exists, trying to checkout ref '%s' for "
+                            "repo %s", target_dir, commit, git_url)
+                repo_commit, repo_depth = reset_git_repo(target_dir, commit, depth)
```

Comment:
`reset_git_repo` does a `git reset --hard`. That should be fine, because in OSBS2 we will only be modifying copies of the repo, not the original, but to be absolutely safe  maybe the reset function could instead raise an error if any changes would be discarded by the hard reset

Reply:
updated to check if there are any local uncommitted changes.